### DISCUSSION TITLE
Is Redis v6.2 really required? Or Redis v6.0+ is enough?

### DISCUSSION THREAD

**Discussion by collimarco** [sidekiq/sidekiq#5835] [Q&A]
I just upgraded all the infrastructure of a web application from Ubuntu 18.04LTS to **Ubuntu 22.04LTS**, which is **the latest LTS available**.

Then I was going to upgrade Sidekiq to the latest version (7).

However I notice that you write that Redis 6.2+ is required.

We plan to keep using Ubuntu 22.04LTS for some years and it uses Redis v6.0.x. 

Does that mean that we can never use Sidekiq v7? Or v6.0 of Redis is enough?

**Comment by mperham** [comment]
Yeah, that's unfortunate. Sidekiq 7 requires Redis 6.2. You can upgrade Redis or stay on Sidekiq 6.5 for a while.

**Reply by collimarco** [reply]
Thanks for the reply.
I think that many people use the LTS versions of Ubuntu / Debian, etc. and they will be in the same situation.
Ubuntu 22.04LTS is the latest LTS available and will be around for a long time.
Are there any important reasons that prevent the support of Redis v6.0 ?

**Reply by mperham** [reply]
Redis 6.2 was released in Mar 2021 and introduced a lot of deprecations and new commands. I felt it was the best version to target. It should be trivial to install 6.2 on any platform using redis.com's package repo:

https://redis.io/docs/getting-started/installation/install-redis-on-linux/

**Reply by collimarco** [reply]
Thanks, I also found that page. Still I see some major risks with that solution: for my understanding, instead of getting a fixed version, I will get the new versions of Redis as soon as they are released. What if they release Redis 8 for example and Sidekiq is not compatible with that? In that case my server would be updated to Redis 8 and cannot be rolled back, while Sidekiq is not yet compatible. I see that as an hazard. For that reason I thought that having a fixed version (the default in Ubuntu, i.e. Redis 6.0) was more stable and reliable.

**Reply by mperham** [reply]
Why not pin a particular version? Can't you say something like "~> 7.0.5" to pick up any Redis 7.0.x APT package but not 7.1?

**Reply by collimarco** [reply]
Ideally yes, but in practice package pinning seems not very common (or well documented). It's not as straightforward and widely used as in the Ruby ecosystem. I would prefer to stick to a standard practice to avoid surprises. 
For how long do you plan to support Sidekiq v6? Is there any EOL?

**Comment by mperham** [comment]
I support the latest two major versions. Once 8 is out, 6 is EOL.
